### ISSUE TITLE
Sidecar/Apple watch unlock not work when using iMacPro1,1

### ISSUE THREAD

**Issue by Leoyzen** [Leoyzen/KVM-Opencore#5]
As far as I know, sidecar need iGPU or T2-chip, but we don't have either.
The workaroud is switch SMBios to iMac18,3 or iMac19,1, then add `shiki-id=Mac-7BA5B2D9E42DDD94 shikigva=40` to boot-args.But may have other problems (I don't test it yet).

**Comment by Leoyzen** [comment]
Someone says it work when using MacPro7,1, but I don't find much information about it. Need more description or investigation.
_Originally posted by @sonoseco in https://github.com/Leoyzen/KVM-Opencore/issues/4#issuecomment-606282631_

**Comment by tpruszin** [comment]
![Screen Shot 2020-04-30 at 5 47 21 pm](https://user-images.githubusercontent.com/1776057/80688308-bfade580-8b0a-11ea-9d42-2b075f45a87c.png) mine is suddenly working for some reason

**Comment by sarkrui** [comment]
From what I know, Apple Watch Unlock works as long as **Continuity** works fine (for example, receiving (or sending) images from (or to) iPhone as well as). This would require you to have a genuine but unused set of SMBIOS info (Serial, MLB, System UUID) by other Apple hardware.

**Comment by tpruszin** [comment]
Can also confirm Sidecar working SMBIOS iMac 19,2 and iPad Pro

**Comment by usr-sse2** [comment]
> From what I know, Apple Watch Unlock works as long as **Continuity** works fine (for example, receiving (or sending) images from (or to) iPhone as well as). This would require you to have a genuine but unused set of SMBIOS info (Serial, MLB, System UUID) by other Apple hardware.

Requirements to Wi-Fi and Bluetooth hardware for Apple Watch unlock are harder than for Continuity.

**Comment by sarkrui** [comment]
> Requirements to Wi-Fi and Bluetooth hardware for Apple Watch unlock are harder than for Continuity.

Understood, and I emphasized explicitly on the bi-directional Continuity. Some hackintosher have reported that they only have unidirectional Continuity. I think bi-directional Continuity is a prerequisite for Apple Watch unlock.

**Comment by heccoder** [comment]
If I can add anything to this is that I was only able to make Apple Watch Unlock work with an original Apple Wifi-BT card, not even the Fenvi card worked for me. Fortunately I had an iMac laying around and I'm using its Wifi-BT card with PCIe adapter. However, Sidecar is still a no go for me but it is probably because I'm using MacPro7,1.

**Comment by sarkrui** [comment]
> However, Sidecar is still a no go for me but it is probably because I'm using MacPro7,1.

In this regard, it's more about the T2 chip issue as it demands HEVC decode for realtime rendering, particularly over Wi-Fi. There's a post from [Sidecar not working on iGPU-less systems reliably](https://github.com/AMD-OSX/bugtracker/issues/1#issuecomment-628928773), reporting that SideCar kinda works on MacPro7,1 on 10.15.5 beta.

**Comment by usr-sse2** [comment]
I have bi-directional Continuity, but there is no Apple Watch unlock option in System Preferences. My Wi-Fi adapter is 802.11n, 2.4 GHz only Broadcom, Bluetooth is 4.0 Broadcom

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sarkrui** [comment]
> I have bi-directional Continuity, but there is no Apple Watch unlock option in System Preferences. 
> 

Hold on, do you have an Apple Watch paired with your iCloud account?

**Comment by usr-sse2** [comment]
> > I have bi-directional Continuity, but there is no Apple Watch unlock option in System Preferences.
> 
> Hold on, do you have an Apple Watch paired with your iCloud account?

Yes, a Series 2. According to Apple, Auto Unlock works starting from 2013 MacBookAir which has Wi-Fi ac, and other Continuity features work starting from 2012 MacBookAir which has 5 GHz n. Strange since Apple Watch doesn't have ac, but it's so. Maybe ac itself is not needed, but Broadcom made some features for distance measurement in the chipset at the same time as ac appeared.

https://support.apple.com/en-us/HT204689

**Comment by Leoyzen** [comment]
> Can also confirm Sidecar working SMBIOS iMac 19,2 and iPad Pro

iMac 19,1/19,2 doesn't have a T2 chip, so it should work as expected.

**Comment by tchlyah** [comment]
I have the same problem. Apple watch unlock was working fine with Clover, but after switching to OpenCore, it doesn't work anymore. I'm using iMac19,1!

![Apple watch](https://user-images.githubusercontent.com/3661112/97028671-bdc65b80-155c-11eb-963c-7a3fc938939d.png)

**Comment by sarkrui** [comment]
@tchlyah  Have you filled the `System Product Name` `System Serial Number` `MLB` `System UUID` for the OC config? That seems crucial for continuity.
![image](https://user-images.githubusercontent.com/14228455/97108083-64c90580-16cb-11eb-8a1d-2fb88e2cbe04.png)

**Comment by tchlyah** [comment]
It turn out it wasn't a Hackintosh problem, I noticed it when I saw that it wasn't working on my Macbook Pro neither! 

I followed this process and it worked like a charm: https://discussions.apple.com/thread/251803595
